### PR TITLE
Add dynamic spectrum synthetic data tutorial

### PR DESCRIPTION
**Pull request by rikvl**

### CODE REVIEW

File: docs/examples/single_screen.rst [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+        cax.set_ylabel('normalized intensity')
+
+
+Setting up a scattering screen
+==============================
+
+Set up the screen by defining angles of the scattering points parallel to
+:math:`\theta_\parallel` and perpendicular to :math:`\theta_\perp` the
+direction of effective velocity of the line of sight along the screen.
+For this example, we choose the one-dimensional screen to be perfectly alligned
```
Comment (mhvk) [inline docs/examples/single_screen.rst:90]: I think it is important to phrase this such that it is clear that for a one-dimensional screen, `theta_perp` has to be zero. E.g.,
```
For this example, we want to mimic a one-dimensional screen, 
in which all points appear to be on a line that intersects with the pulsar. 
Hence, we set the perpendicular angles to zero.
```

File: docs/examples/single_screen.rst [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+    :include-source:
+    :context:
+    
+    th_par_t = th_par[:,np.newaxis,np.newaxis] + mu_eff * t
+    theta_t_squared = th_par_t**2 + th_perp[:,np.newaxis,np.newaxis]**2
+    tau_t = (((d_eff / (2*const.c)) * theta_t_squared)
+             .to(u.s, equivalencies=u.dimensionless_angles()))
+
+    phasor = np.exp(1j * (f * tau_t * u.cycle).to_value(u.rad))
+    efields = phasor * magnification[:,np.newaxis,np.newaxis]
```
Comment (mhvk) [inline docs/examples/single_screen.rst:214]: Total nitpick, but add spaces after the commas.

**Review by mhvk — COMMENTED** [review]
Looks very good modulo a few small comments.

File: docs/examples/single_screen.rst [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+Have a look at the electric fields associated with the individual scattered
+images.
+
+.. plot::
+    :include-source:
+    :context: close-figs
+
+    fig, axes = plt.subplots(nrows=2, ncols=2, figsize=(12., 8.))
+    plt.subplots_adjust(wspace=0.4, hspace=0.4)
+    for i in range(efields.shape[0]):
```
[the remark this answers is not in the document]
Comment (rikvl) [reply docs/examples/single_screen.rst]: Implemented.

File: docs/examples/single_screen.rst (hunk 1 above)
Comment (rikvl) [reply docs/examples/single_screen.rst]: I changed things to your earlier suggestion of leaving out the `theta_perp` altogether. This bit of text is added in the note about using the `dynamic_field` function, where `theta_perp` is used

File: docs/examples/single_screen.rst (hunk 2 above)
Comment (rikvl) [reply docs/examples/single_screen.rst]: Done. I appreciate the nitpick!

**Comment by rikvl** [comment]
@mhvk -- I implemented the comments and made some further small changes

**Review by mhvk — APPROVED** [review]
Really nice now!  Let's get it in...

**Pull request merged**